Makefile

```diff
@@ -0,0 +1,283 @@
+gcloud := $(shell which gcloud) --quiet --verbosity=error
+git := $(shell which git)
+go := $(shell which go)
+grep := $(shell which grep) --quiet --no-messages
+kubectl := $(shell which kubectl)
+terraform := $(shell which terraform)
+
+# project options
+project_tmp_dir = tmp
+
+# plugin's binary options
+binary_name = drone-gke
+coverage_name = coverage.out
+
+# git options
+git_default_branch = $(shell cat $(CURDIR)/.git/refs/remotes/origin/HEAD | cut -d ' ' -f 2 | cut -d '/' -f 4)
+git_current_branch = $(shell $(git) rev-parse --abbrev-ref HEAD)
+git_current_revision = $(shell $(git) rev-parse --short HEAD)
+
+# plugin's docker options
+docker_default_repo_name = nytimes
+docker_image_name = drone-gke
+docker_default_tag = latest
+ifneq ($(git_current_branch), $(git_default_branch))
+	# if current branch name is NOT .git/refs/remotes/origin/HEAD (i.e., master)
+	# then use the current branch name as the default docker tag
+	docker_default_tag = $(git_current_branch)
+endif
+docker_default_file = Dockerfile
+docker_default_build_context = $(CURDIR)
+
+# gcloud SDK options
+gcloud_default_sdk_tag = alpine
+
+# test resource options
+test_sa_key_path = $(project_tmp_dir)/key.json
+terraform_base_dir = terraform
+terraform_dir = $(terraform_base_dir)/.terraform
+terraform_state_file = $(terraform_base_dir)/terraform.tfstate
+terraform_plan_file = $(terraform_base_dir)/test.tfplan
+
+# print usage (default target)
+.PHONY : usage
+usage :
+	@echo "Usage:"
+	@echo ""
+	@echo "  \$$ make check               # run tests for required software"
+	@echo "  \$$ make clean               # delete plugin binary, docker image, test resources"
+	@echo "  \$$ make release             # build and push plugin docker image"
+	@echo "  \$$ make docker-image        # build plugin docker image"
+	@echo "  \$$ make docker-push         # push plugin docker image"
+	@echo "  \$$ make run                 # run docker container using local-example"
+	@echo "  \$$ make $(binary_name)           # build plugin binary"
+	@echo "  \$$ make $(coverage_name)        # test plugin binary"
+	@exit 0
+
+# check that docker is installed
+.PHONY : check-docker
+check-docker :
+ifneq (0,$(shell command -v $(docker) 2>&1 >/dev/null ; echo $$?))
+	$(error "docker not installed")
+else
+	@exit 0
+endif
+
+# check that gcloud is installed
+.PHONY : check-gcloud
+check-gcloud :
+ifneq (0,$(shell command -v $(gcloud) 2>&1 >/dev/null ; echo $$?))
+	$(error "gcloud not installed")
+else
+	@exit 0
+endif
+
+# check that git is installed
+.PHONY : check-git
+check-git :
+ifneq (0,$(shell command -v $(git) 2>&1 >/dev/null ; echo $$?))
+	$(error "git not installed")
+else
+	@exit 0
+endif
+
+# check that go is installed
+.PHONY : check-go
+check-go :
+ifneq (0,$(shell command -v $(go) 2>&1 >/dev/null ; echo $$?))
+	$(error "go not installed")
+else
+ifneq (0,$(shell $(go) version | cut -d ' ' -f 3 | tr -d 'go' | $(grep) --extended '^1\.12' ; echo $$?))
+	$(error "go version 1.12.x is required")
```

Comment:
This will fail with Go 1.13beta1 or higher. I also think drone-gke should work just fine with Go 1.11, so Go 1.12 isn't actually required.

I'm not sure we should check the version of Go, should we remove this check? x) Or just check if Go is installed, without checking the version.

Reply:
updated to remove the version check..